include/aie/Dialect/AIE/IR/AIETargetModel.h

```diff
@@ -301,6 +316,10 @@ class AIETargetModel {
   // their corresponding lengths in bytes (second).
   virtual std::vector<std::pair<uint32_t, uint32_t>>
   getShimBurstEncodingsAndLengths() const = 0;
+
+  // Returns a bfpType with the corresponding caracteristics or std::nullopt if
+  // unsupported.
+  virtual std::optional<BfpType> getBfpType(std::string bfpName) const = 0;
```

Comment:
I think type "getters" should be part of the type or dialect interface, e.g.:
```c++
AIEX::BlockFloatType AIEX::BlockFloatType::get(std::string &bfpName) {
 ...
}
```
In the TargetModel I would expect something more like
```
bool TargetModel::isSupported(mlir::Type t) { 
  ...
}
```

Reply:
The refactored code looks good to me. The string interface avoids the dependency issue and could be cleaned up later if/when the type is moved to AIE dialect.